part2/stages/Make-ESP-partition

```diff
@@ -0,0 +1,61 @@
+#!/bin/ash
+#
+# Copyright (c) 2013 Citrix Systems, Inc.
+# 
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+. ${SCRIPT_DIR}/functions
+. ${DISK_CONF}
+. ${PARTITION_CONF}
+. ${SCRIPT_DIR}/Functions/install-main
+
+create_efi_system_partition()
+{
+    local PARTITION="${XC_PARTITION}"
+    local DISK=$(get_partition_disk "${PARTITION}")
+    local DISK_DEV="/dev/${DISK}"
+
+    local ESP=$(sfdisk -d ${DISK_DEV} | grep "type=C12A7328-F81F-11D2-BA4B-00A0C93EC93B" | awk '{ print $1 }')
```

Comment:
You could use a variable, EFI_SYSTEM_PARTITION_GUID, to make clear what you are checking.